concise_concepts/conceptualizer/Conceptualizer.py

```diff
@@ -357,17 +329,18 @@ def add_patterns(input_dict):
                     words = input_dict[key]
                 for word in words:
                     if word != key:
-                        specific_match_rule = dict()
-                        specific_match_rule.update(self.match_rule)
-                        word_parts = re.split(f"[{self.word_delimiter}]+", word)
+                        specific_match_rule = {**self.match_rule}
+                        word_parts = re.split(
+                            f"[{re.escape(self.word_delimiter)}]+", word
+                        )
```

Comment:
Use of `re.escape` to prevent weirdness if someone for whatever reason chooses `self.word_delimiter = "\"` or `"]"`.

Reply:
great, I think I did already use this somewhere else right?

Reply:
You did indeed!